Document selective context (#584)
preactjs/preact-www: content/en/guide/v10/hooks.md

}
```

> The `useContext` hook accepts a second arguments, we call this the `shouldUpdate` function, this will be registered once and will determine wheter or not your component should rerender.

Reviewer: ```suggestion
The `useContext` hook accepts an optional `shouldUpdate` function as a second argument. This function will be called each time the context value is changed, and can return a Boolean value indicating whether the subscribed component should rerender.
```

```jsx
const Theme = createContext({ theme: 'light' });

function DisplayTheme() {

>   const theme = useContext(Theme, (ctx, prevCtx) => ctx.theme !== prevCtx.theme);

Reviewer: ```suggestion
  const { theme } = useContext(Theme, (ctx, prevCtx) => ctx.theme !== prevCtx.theme);
```

  return <p>Active theme: {theme}</p>;
}

// ...later
function App() {
  return (
    <Theme.Provider value={{ theme: 'light' }}>
      <OtherComponent>
        <DisplayTheme />
      </OtherComponent>
    </Theme.Provider>
  )
}
```

> In the above example we'll only rerender if the current context and previous context are different from one another.

Reviewer: ```suggestion
In the above example, the custom `shouldUpdate` function means `DisplayTheme` will only be re-rendered when the context object's `.theme` property is changed.
```

> > Note that using a conditional there like `condition ? func1 : func2` will only use the function from the first value since it doesn't
>   track the need for re-registering.

Reviewer: ```suggestion
> Note: the `shouldUpdate` function is registered once and can't be changed dynamically. New function references passed as the second argument to useContext will be ignored.
```
